packages/sommelier.rb

```diff
@@ -56,7 +49,9 @@ def self.preflight
     when 'i686', 'x86_64'
       @container_check = `/usr/bin/crossystem inside_vm` == '1'
     end
-    abort 'This package is not compatible with your device :/'.lightred unless File.socket?('/var/run/chrome/wayland-0') || @container_check
+    return if File.socket?('/var/run/chrome/wayland-0') || @container_check
+
+    abort 'This package is not compatible with your device :/'.lightred
```

Comment:
Would this logic be better?
case ARCH
    when 'armv7l', 'aarch64'
      @container_check = File.exist?('/.dockerenv')
    when 'i686', 'x86_64'
      @container_check = `/usr/bin/crossystem inside_vm` == '1'
    else
      abort 'This package is not compatible with your device :/'.lightred
    end
    return if File.socket?('/var/run/chrome/wayland-0') || @container_check
```

Reply:
The suggestion is this:
```
  def self.preflight
    case ARCH
    when 'armv7l', 'aarch64'
      @container_check = File.exist?('/.dockerenv')
    when 'i686', 'x86_64'
      @container_check = /usr/bin/crossystem inside_vm == '1'
    else
      abort 'This package is not compatible with your device :/'.lightred
    end
    return if File.socket?('/var/run/chrome/wayland-0') || @container_check
    
  end
```

The abort happens if ARCH isn't armv7l, aarch64, x86_64, i686. That isn't useful because that will always be false.  We want the abort to happen if return doesn't happen.

Reply:
Oh, so this would abort for i686 then?